src/formatter/junit_formatter_spec.ts

```diff
@@ -0,0 +1,245 @@
+import { afterEach, beforeEach, describe, it } from 'mocha'
+import { expect, use } from 'chai'
+import chaiXml from 'chai-xml'
+import FakeTimers, { InstalledClock } from '@sinonjs/fake-timers'
+import timeMethods from '../time'
+import { testFormatter } from '../../test/formatter_helpers'
+import { getJsonFormatterSupportCodeLibrary } from '../../test/fixtures/json_formatter_steps'
+
+use(chaiXml)
+
+describe('JunitFormatter', () => {
```

Comment:
Good testing here, there are some cases you'll want to cover as well though:

- pending steps
- skipped test cases
- scenario outline with multiple examples